src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -445,6 +477,29 @@ private sealed class DataAndUShortEqualityComparer : EqualityComparer<(Immutable
             public override int GetHashCode((ImmutableArray<byte> Data, ushort Value) obj) =>
                 ByteSequenceComparer.GetHashCode(obj.Data); // purposefully not including Value, as it won't add meaningfully to the hash code
         }
+
+        private sealed class ConstantValueAndUShortEqualityComparer : EqualityComparer<(ImmutableArray<ConstantValue> Constants, ushort Value)>
+        {
+            public static readonly ConstantValueAndUShortEqualityComparer Instance = new ConstantValueAndUShortEqualityComparer();
+
+            private ConstantValueAndUShortEqualityComparer() { }
+
+            public override bool Equals((ImmutableArray<ConstantValue> Constants, ushort Value) x, (ImmutableArray<ConstantValue> Constants, ushort Value) y) =>
+                x.Value == y.Value &&
+                ByteSequenceComparer.Equals(x.Constants, y.Constants);
```

Comment:
>ByteSequenceComparer.Equals(x.Constants, y.Constants); [](http://example.com/codeflow?start=16&length=54)

I am not sure what method is called here. Could you clarify please?

Reply:
That was just wrong :-/ We're dealing with `ConstantValue` not bytes here. This was definitely not doing what I intended... Thanks for catching this.